Sources/BraintreeCore/BTAnalyticsService.m

```diff
@@ -241,27 +240,6 @@ - (void)flush:(void (^)(NSError *))completionBlock {
     }];
 }
 
-#pragma mark - Private methods
-
-- (void)appWillResign:(NSNotification *)notification {
-    UIApplication *application = notification.object;
```

Comment:
qq: Are analytics all processed in the foreground now?

Reply:
We didn't change the thread where analytics are being processed which is done in the `sendAnalyticsEvent` function. Here we were unnecessarily flushing analytics on the background thread and updating this results in all of the analytics events being sent as expected, so was unnecessarily adding strain.

Reply:
Yeah background tasks do still run when the app is closed by the user, it may be good to do this in the future. On Android we're using `WorkManager` to do timed uploads every 30 seconds, I'm wondering if iOS has something similar.

Reply:
Yeah, previously we were getting a _ton_ of flush analytics errors which this resolves. I think we should certainly look into something similar in the future. When we re-write the core module in Swift I think there is certainly a lot of optimization to be done as we move to newer APIs with those changes in the future.